github/Variable.py

```diff
@@ -0,0 +1,149 @@
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+import github
+import github.GithubObject
+import github.Repository
+
+
+class Variable(github.GithubObject.CompletableGithubObject):
+    """
+    This class represents a GitHub variable. The reference can be found here https://docs.github.com/en/rest/actions/variables
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"name": self.name})
+
+    @property
+    def name(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._name)
+        return self._name.value
+
+    @property
+    def value(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self.value)
+        return self._value.value
+
+    @property
+    def created_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._created_at)
+        return self._created_at.value
+
+    @property
+    def updated_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._updated_at)
+        return self._updated_at.value
+
+    @property
+    def visibility(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._visibility)
+        return self._visibility.value
+
+    @property
+    def selected_repositories(self):
+        """
+        :calls: `GET {secret_url}/repositories <https://docs.github.com/en/rest/actions/variables#list-selected-repositories-for-an-organization-secret>`_
+        :type: List of type Repository
+        """
+        self._completeIfNotSet(self._selected_repositories)
+        return self._selected_repositories.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        return self._url.value
+
+    def delete(self):
+        """
+        :calls: `DELETE {variable_url} <https://docs.github.com/en/rest/actions/variables>`_
+        :rtype: None
+        """
+        headers, data = self._requester.requestJsonAndCheck("DELETE", self.url)
+
+    def add_repo(self, repo):
```

Comment:
Maybe note that this only works for organization variables with "selected" visibility, not for repo variables.

Reply:
Alternatively, we could have `class OrganizationVariable(Variable)`, which has `add_repo` and `remove_repo`, and `Organization.get_variable` returns `OrganizationVariable`.